### PR TITLE
Develop #153

### PR DESCRIPTION
**Pull request by rpitingolo**
@lhendey making progress. All of the vars are defined from the source data. I still have some percentages to create. You can take a look now or after I finish the percentages tomorrow. 

I don't think we can do the disability tables (T6) as it's one of the ones not available at tract-level. The previous program didn't read in T6.

### CODE REVIEW

**Comment by lhendey** [comment]
@rpitingolo  - what's this about? FILENAME MPRINT 'd:\dcdata\MACCODE.SAS';

**Comment by lhendey** [comment]
@rpitingolo are you outputting to the main HUD library rather than hud/chas/prog/ subfolder? 

D:\DCData\Libraries\HUD\&level\?

I need you to use the macro var for the default directory because i have the c drive.

**Comment by lhendey** [comment]
@rpitingolo = apologies - apparently stuck in old program. you have the summary indicators in the macro - but am i correct there's no program to run the macro yet?

File: Macros/CHAS_summary_vars.sas [hunk 1]
```diff
@@ -0,0 +1,964 @@
...
+			owner_2br_tot_&years. = "Owner-occupied 2 bedroom housing units, &years_dash."
+			owner_3plusbr_tot_&years. = "Owner-occupied 3+ bedroom housing units, &years_dash."
+			owner_01br_aff50_&years. = "Owner-occupied 0-1 bedroom housing units affordable at 50% AMI, &years_dash."
+			owner_2br_aff50_&years. = "Owner-occupied 2 bedroom housing units affordable at 50% AMI, &years_dash."
+			owner_3plusbr_aff50_&years. = "Owner-occupied 3+ bedroom housing units affordable at 50% AMI, &years_dash."
+			owner_allbr_aff50_&years. = "Owner-occupied housing units affordable at 50% AMI, &years_dash."
+			;
+
+	%Pct_calc( var=Powner_units_tot, label=% Owner-occupied housing units, num=owner_units_tot, den=all_units_tot, years= &years. );
+	%Pct_calc( var=Pforsale_units_tot, label=% Vacant housing units for sale, num=forsale_units_tot, den=all_units_tot, years= &years. );
```
Comment (lhendey) [inline Macros/CHAS_summary_vars.sas]: denominator should be (owner occupied + vacant for sale)

File: Macros/CHAS_summary_vars.sas [hunk 2]
```diff
@@ -0,0 +1,964 @@
...
+	%Pct_calc( var=Prenter_unit_tot, label=% Renter-occupied housing units, num=renter_unit_tot, den=all_units_tot, years= &years. );
```
Comment (lhendey) [inline Macros/CHAS_summary_vars.sas]: denominator for % vacant for rent should be (renter occupied + vacant for rent)

File: Macros/CHAS_summary_vars.sas [hunk 3]
```diff
@@ -0,0 +1,964 @@
...
+	%Pct_calc( var=Pforrent_units_tot, label=% Vacant housing units for rent, num=forrent_units_tot, den=all_units_tot, years= &years. );
+
+	%Pct_calc( var=Prenter_unit_aff30, label=% Renter-occupied housing units affordable at 30% AMI, num=renter_unit_aff30, den=tot_aff30, years= &years. );
```
Comment (lhendey) [inline Macros/CHAS_summary_vars.sas]: denom = renter occupied units for both of these

File: Macros/CHAS_summary_vars.sas [hunk 4]
```diff
@@ -0,0 +1,964 @@
...
+	%Pct_calc( var=Pforrent_units_aff30, label=% For-rent housing units affordable at 30% AMI, num=forrent_units_aff30, den=tot_aff30, years= &years. );
+
+	%Pct_calc( var=Prenter_unit_aff50, label=% Renter-occupied housing units affordable at 50% AMI, num=renter_unit_aff50, den=tot_aff50, years= &years. );
```
Comment (lhendey) [inline Macros/CHAS_summary_vars.sas]: denom = renter occupied units for both of these

File: Macros/CHAS_summary_vars.sas [hunk 5]
```diff
@@ -0,0 +1,964 @@
...
+	%Pct_calc( var=Pforrent_units_aff50, label=% For-rent housing units affordable at 50% AMI, num=forrent_units_aff50, den=tot_aff50, years= &years. );
+
+	%Pct_calc( var=Powner_unit_aff50, label=% Owner-occupied housing units affordable at 50% AMI, num=owner_unit_aff50, den=tot_aff50, years= &years. );
```
Comment (lhendey) [inline Macros/CHAS_summary_vars.sas]: denom = owner occupied units for these two.

File: Macros/CHAS_summary_vars.sas [hunk 6]
```diff
@@ -0,0 +1,964 @@
...
+			owner_01br_tot_&years. = "Owner-occupied 0-1 bedroom housing units, &years_dash."
+			owner_2br_tot_&years. = "Owner-occupied 2 bedroom housing units, &years_dash."
+			owner_3plusbr_tot_&years. = "Owner-occupied 3+ bedroom housing units, &years_dash."
+			owner_01br_aff50_&years. = "Owner-occupied 0-1 bedroom housing units affordable at 50% AMI, &years_dash."
+			owner_2br_aff50_&years. = "Owner-occupied 2 bedroom housing units affordable at 50% AMI, &years_dash."
+			owner_3plusbr_aff50_&years. = "Owner-occupied 3+ bedroom housing units affordable at 50% AMI, &years_dash."
+			owner_allbr_aff50_&years. = "Owner-occupied housing units affordable at 50% AMI, &years_dash."
+			;
+
+	%Pct_calc( var=Powner_units_tot, label=% Owner-occupied housing units, num=owner_units_tot, den=all_units_tot, years= &years. );
```
Comment (lhendey) [inline Macros/CHAS_summary_vars.sas]: denom = occupied units

File: Macros/CHAS_summary_vars.sas (hunk 2 above)
Comment (lhendey) [inline Macros/CHAS_summary_vars.sas]: denom for % renter occupied is all occupied units

File: Macros/CHAS_summary_vars.sas [hunk 7]
```diff
@@ -0,0 +1,964 @@
...
+	%Pct_calc( var=Pforrent_units_aff50, label=% For-rent housing units affordable at 50% AMI, num=forrent_units_aff50, den=tot_aff50, years= &years. );
+
+	%Pct_calc( var=Powner_unit_aff50, label=% Owner-occupied housing units affordable at 50% AMI, num=owner_unit_aff50, den=tot_aff50, years= &years. );
+	%Pct_calc( var=Pforsale_units_aff50, label=% For-sale housing units affordable at 50% AMI, num=forsale_units_aff50, den=tot_aff50, years= &years. );
+
+	%Pct_calc( var=Prenter_01br_tot, label=% Renter-occupied 0-1 bedroom housing units, num=renter_01br_tot, den=renter_unit_tot, years= &years. );
+	%Pct_calc( var=Prenter_2br_tot, label=% Renter-occupied 2 bedroom housing units, num=renter_2br_tot, den=renter_unit_tot, years= &years. );
+	%Pct_calc( var=Prenter_3plusbr_tot, label=% Renter-occupied 3+ bedroom housing units, num=renter_3plusbr_tot, den=renter_unit_tot, years= &years. );
+
+	%Pct_calc( var=Prenter_01br_aff30, label=% Renter-occupied 0-1 bedroom housing units affordable at 30% AMI, num=renter_01br_aff30, den=renter_unit_aff30, years= &years. );
```
Comment (lhendey) [inline Macros/CHAS_summary_vars.sas]: I'm probably also going to want both row % and column % for these. 

so num = renter_01br_aff30 and denom = renter_01br_tot in addition to what you have.

**Comment by rpitingolo** [comment]
@lhendey going through the line comments but yes, for testing purposes you can run the datastep in the macro. I'll add a separate program in the /prog folder at the end.

**Comment by lhendey** [comment]
@rpitingolo can you continue to move forward on this before I finish the review?

**Comment by rpitingolo** [comment]
@lhendey I'm working on it now. Will tag again when it's ready.

**Comment by rpitingolo** [comment]
@lhendey I finished the percentages. Hopefully interpreted the row/column percents correctly and used the right denominators.

**Comment by lhendey** [comment]
@rpitingolo okay - will try to review in the morning.

**Comment by rpitingolo** [comment]
@lhendey thanks!

File: Macros/CHAS_summary_vars.sas (hunk 7 above)
Comment (lhendey) [reply Macros/CHAS_summary_vars.sas]: @rpitingolo i'm looking through the rest - but it doesn't look like this part was done?

**Comment by rpitingolo** [comment]
@lhendey thanks for review. I'm looking at your bullets now.

**Comment by rpitingolo** [comment]
@lhendey I updated the Excel workbook on Box to reflect the denominator edits. Can you please look at that and confirm it's right before I update the code?

I'll check into the disability tables.

**Comment by lhendey** [comment]
@rpitingolo workbook looks good - left a comment in the affordability tab.

**Comment by rpitingolo** [comment]
@lhendey I see the comment and you are right, I forgot that those categories aren't mutually exclusive. I'll update the code now.

**Comment by rpitingolo** [comment]
@lhendey I updated the denominators if you want to take another pass through. I can start prepping the program to output the tables as you do that.

**Comment by lhendey** [comment]
@rpitingolo the demand size still have persons per room as the denominator instead of income groups?
what did you learn about the disability?

**Comment by rpitingolo** [comment]
@lhendey I double checked and the disability indicators (table 6) are only at the county-level.

**Comment by rpitingolo** [comment]
@lhendey heads-up that I am planning to have the output program ready to review in the morning.

**Comment by rpitingolo** [comment]
@lhendey Output_CHAS_Tables is ready for review.

**Comment by lhendey** [comment]
@rpitingolo you have the output going to the raw folder?

**Comment by lhendey** [comment]
@rpitingolo we should save the national files (sans drop vars) and register them in them in the metadata.

**Comment by rpitingolo** [comment]
@lhendey I can move the output to the prog folder. It's just updating the let statement.

**Comment by lhendey** [comment]
@rpitingolo do you have an excel workbook set up with the csv's pulled in?

**Comment by rpitingolo** [comment]
@lhendey I am prepping the Tables from CHAS.xlsx on Box right now.

**Comment by lhendey** [comment]
@rpitingolo - having trouble checking this-  the output doesn't have any var names or labels. i'll try to look at the reshape macro.

**Comment by rpitingolo** [comment]
@lhendey the reshape macro takes the vars in the parameter and assembles them into columns 1,2,etc. to match the table shells. Since the summary vars macro creates unique variable names, if you try to simply set data sets with a unique keep statement it retains all the columns. So instead of a 4x4 table you would get a 4x16 table with a bunch of missing cells, if that makes sense. 

In the old days we probably would have used a DDE inside a macro to export these into the correct cells in the workbook. That's not possible anymore (doesn't work on my system at least) and this was the closest to that I could come up with.

**Comment by rpitingolo** [comment]
@lhendey the workbook now pulls in the data from the CSVs. I haven't thoroughly checked yet but all of the percents add up to 100% where they are supposed to, so that's a good start. 
https://urbanorg.app.box.com/file/583069175999

**Comment by rpitingolo** [comment]
@lhendey I saved a county workbook and place workbook, both here: https://urbanorg.app.box.com/folder/92903192078

**Comment by lhendey** [comment]
@rpitingolo merge and close this issue and do any follow-up work on new issues/branches.

**Pull request merged**